src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -2516,18 +2480,23 @@ Status Session::receive_error_message(const ProtocolErrorInfo& info)
         return {ErrorCodes::SyncProtocolInvariantFailed, "Received ERROR message when it was not legal"};
     }
 
-    bool known_error_code = bool(get_protocol_error_message(info.raw_error_code));
-    if (REALM_UNLIKELY(!known_error_code)) {
-        return {ErrorCodes::SyncProtocolInvariantFailed,
-                util::format("Received ERROR message with unknown error code %1", info.raw_error_code)};
+    auto status = protocol_error_to_status(static_cast<ProtocolError>(info.raw_error_code), info.message);
+    // If the server has requested an action but sent an invalid error code we can still process the error and pass an
+    // UnknownError status to the user. This lets us add error codes on the server side without strictly having to
+    // bump the protocol version to avoid crashing older clients.
+    if (status == ErrorCodes::UnknownError && info.server_requests_action != ProtocolErrorInfo::Action::NoAction) {
+        return {
+            ErrorCodes::SyncProtocolInvariantFailed,
+            util::format("Received ERROR message with unknown error code %1 and no action.", info.raw_error_code)};
```

Comment:
an action **is** received in this case

Reply:
Yup, this logic is wrong I've undone it and will fix it up in a separate review.